dataland-frontend/tests/e2e/specs/sfdr/DataUpload.ts

```diff
@@ -0,0 +1,106 @@
+      cy.get(".p-datatable-tbody").find(".p-rowgroup-header").eq(0).should("have.text", "General");
+
+      cy.get(".p-datatable-tbody").find(".p-rowgroup-header").eq(2).should("have.text", "Biodiversity");
+      toggleRowGroup("biodiversity");
+      cy.contains("td.headers-bg", "Primary Forest And Wooded Land Of Native Species Exposure").should("exist");
+      cy.contains("td.headers-bg", "Primary Forest And Wooded Land Of Native Species Exposure")
+        .siblings("td")
+        .should("have.text", "Yes");
+      cy.contains("td.headers-bg", "Protected Areas Exposure").should("exist");
+      cy.contains("td.headers-bg", "Protected Areas Exposure").siblings("td").should("have.text", "No");
+      cy.contains("td.headers-bg", "Rare Or Endangered Ecosystems Exposure").should("exist");
+      cy.contains("td.headers-bg", "Rare Or Endangered Ecosystems Exposure").siblings("td").should("have.text", "Yes");
+    }
+
+    /**
+     * Set the quality for the sfdr test dataset
+     */
+    function setQualityInSfdrUploadForm(): void {
+      Cypress.Keyboard.defaults({
+        keystrokeDelay: 0,
+      });
+
+      cy.get('[data-test="primaryForestAndWoodedLandOfNativeSpeciesExposure"]')
+        .find('select[name="quality"]')
+        .select(3);
+      cy.get('[data-test="protectedAreasExposure"]').find('select[name="quality"]').select(3);
+      cy.get('[data-test="rareOrEndangeredEcosystemsExposure"]').find('select[name="quality"]').select(3);
+    }
+    it("Create a company via api and upload a SFDR dataset via the api", () => {
+      const uniqueCompanyMarker = Date.now().toString();
+      const testCompanyName = "Company-Created-In-DataJourney-Form-" + uniqueCompanyMarker;
+      getKeycloakToken(admin_name, admin_pw).then((token: string) => {
+        return uploadCompanyViaApi(token, generateDummyCompanyInformation(testCompanyName)).then((storedCompany) => {
+          return uploadOneSfdrDataset(token, storedCompany.companyId, "2021", testSfdrCompany.t).then(
+            (dataMetaInformation) => {
+              cy.intercept("**/api/companies/" + storedCompany.companyId).as("getCompanyInformation");
+              cy.visitAndCheckAppMount(
+                "/companies/" +
+                  storedCompany.companyId +
+                  "/frameworks/" +
+                  DataTypeEnum.Sfdr +
+                  "/upload" +
+                  "?templateDataId=" +
+                  dataMetaInformation.dataId,
+              );
+              cy.wait("@getCompanyInformation", { timeout: Cypress.env("medium_timeout_in_ms") as number });
+
+              cy.url().should(
+                "eq",
+                getBaseUrl() +
+                  "/companies/" +
+                  storedCompany.companyId +
+                  "/frameworks/" +
+                  DataTypeEnum.Sfdr +
+                  "/upload" +
+                  "?templateDataId=" +
+                  dataMetaInformation.dataId,
+              );
```

Comment:
This url check is not really necessary in my opinion.
The wait on "getCompanyInformation" should already guarantee that the page is actually loading.

Reply:
Removed